mautic/mautic
# Fix: Initialize AssetService properly in GrapesJsBuilder

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | 🟢 
| New feature/enhancement? (use the a.x branch)      | 🔴
| Deprecations?                          | 🔴
| BC breaks? (use the c.x branch)        | 🔴
| Automated tests included?              | 🔴 
| Related user documentation PR URL      | mautic/user-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation-new#... 
| Issue(s) addressed                     | Fixes #14299  



## Description

This PR resolves a runtime error (`TypeError: h.getAssetsConfig is not a function`) in the GrapesJsBuilder plugin caused by directly referencing `AssetService.getAssetsConfig()` without proper initialization.

Replaced the initialization of `BuilderService` in `builder.js` with explicit initialization of `AssetService`:
```
const assetService = new AssetService();
const builder = new BuilderService(assetService);
```

Thanks to @patrykgruszka for pointing out the correct initialization approach. https://github.com/mautic/mautic/pull/14028#discussion_r1863014994






---
### 📋 Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Navigate to **Emails**
3. Click the **New** button
4. Select any email theme
5. Open the email builder, edit some text, and then close the email builder
6. Save the email

It should not throw a JavaScript error:
`Uncaught TypeError: h.getAssetsConfig is not a function.`

![image](https://github.com/user-attachments/assets/b59919b6-d82a-495d-8257-d4e2d1231af0)

https://github.com/user-attachments/assets/a1056d7b-fe9e-4722-b8e6-8cff39a7fb73
